sdk/storage/Azure.Storage.Blobs.ChangeFeed/src/ChangeFeed.cs

```diff
@@ -133,20 +133,23 @@ internal class ChangeFeed
 
         public bool HasNext()
         {
-            if (_empty)
+            // [If Change Feed is empty], or [current segment is not finalized]
+            // or ([segment count is 0] and [year count is 0] and [current segment doesn't have next])
+            if (_empty
+                || !_currentSegment.Finalized
+                || _segments.Count == 0
+                    && _years.Count == 0
+                    && !_currentSegment.HasNext())
             {
                 return false;
             }
 
-            // We have no more segments, years, and the current segment doesn't have hext.
-            if (_segments.Count == 0 && _years.Count == 0  && !_currentSegment.HasNext())
+            if (_endTime.HasValue)
             {
-                return false;
+                return _currentSegment.DateTime <= _endTime;
```

Comment:
I thought we agreed on `_currentSegment.DateTime < _endTime`.
Same in the `GetPage()`.

Reply:
I need to apply this change, thanks for the heads up.

Reply:
Fixed.

Reply:
fixed.